File: code/1008/1008.cpp
/****************************************************
 > @Time       : 2019-10-29
 > @Author     : HuShengQuan
 > @File       : 1008.cpp
 > @Software   : vim
 > @type       : pat practice
*****************************************************/
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    vector<int> num(n);

    for (int i = 0; i < n; i++) {
        cin >> num[i];
    }

    m = m % n;
    if (m != 0) {
        reverse(begin(num), begin(num) + n);
        reverse(begin(num), begin(num) + m);
        reverse(begin(num) + m, begin(num) + n);
    }

    cout << num[0];
    for (int i = 1; i < n; i++) {
        cout << " " << num[i];
    }

    return 0;
}

File: code/1008/1008.go
/*
**@Time       : 2019-10-29
**@Author     : HuShengQuan
**@File       : 1008.go
**@Software   : vim
**@type       : pat practice
*/
package main

import "fmt"

func main() {
    var n int
    var m int
    fmt.Scanf("%d%d", &n, &m)
    A := make([]int, n)

    for i := 0; i < n; i++ {
        num := 0
        fmt.Scanf("%d", &num)
        A[i] = num
    }

    m = m % n
    S := append(A[n - m:], A[:n - m]...)

    fmt.Printf("%d", S[0])
    for i := 1; i < n; i++ {
        fmt.Printf(" %d", S[i])
    }
}
